### PR TITLE
Add new command to create blank, web autostep projects. 

### PR DESCRIPTION
**Pull request by SivaGudivada**
Added a new command line option. 
* AutoStepCommand: pass the properly disposed loggerFactory to `Execute()` method.

### CODE REVIEW

File: src/autostep/NewProjectCommand.cs [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace AutoStep.CommandLine
+{
+    public class NewProjectCommand : AutoStepCommand<RunArgs>
```
Comment (alistairjevans) [inline src/autostep/NewProjectCommand.cs]: As a rule, there should only be one top-level class per-file (among other things, makes it easier to review). Can you split the file out?

File: src/autostep/Program.cs [hunk 2]
```diff
@@ -46,5 +47,13 @@ static async Task<int> Main(string[] args)
...
             return await result.InvokeAsync();
         }
+
+        private static bool DisplayHelp(string[] args) =>
+            NoArgsPassed(args) ||
+            NewCmdWithNoSubCmds(args);
+
+        private static bool NoArgsPassed(string[] args) => args.Length == 0;
+
+        private static bool NewCmdWithNoSubCmds(string[] args) => args.Length == 1 && args[0] == "new";
```
Comment (alistairjevans) [inline src/autostep/Program.cs]: This introduces coupling to the ``NewProjectCommand`` indirectly via it's command word. Could we update ``AutoStepCommand`` to allow a base 'command is valid' method perhaps?

File: src/autostep/Program.cs [hunk 3]
```diff
@@ -46,5 +47,13 @@ static async Task<int> Main(string[] args)
 
             return await result.InvokeAsync();
         }
+
+        private static bool DisplayHelp(string[] args) =>
```
Comment (alistairjevans) [inline src/autostep/Program.cs]: The naming of this method implies action (``Display``), rather than a check. Perhaps rename to ``ShouldDisplayHelp``?

**Review by alistairjevans — CHANGES_REQUESTED** [review]
Looks good initially, but it's hard to review with the multi-class file; could you please split out the big file and I'll continue the review?

Also, I realised I didn't have TreatWarningsAsErrors on for the pull request build; any build warnings should really fail the build on a pull request. I've fixed that, and corrected all the outstanding build warnings in ``develop``, could you rebase?

Thanks

File: src/autostep/NewProjectCommand.cs (hunk 1 above)
Comment (SivaGudivada) [reply src/autostep/NewProjectCommand.cs]: Will do 👍

File: tests/autostep-cli.tests/CommandInvocationTests.cs [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+using Xunit;
+using FluentAssertions;
+using static AutoStep.CommandLine.CreateProjectCommand;
+using FluentAssertions.Primitives;
+
+namespace AutoStep.CommandLine.Tests
+{
+    public class CommandInvocationTests
+    {
+        private static readonly CreateProjectDelegate _successProjectCreation = MakeCreateProjectDelegateFromStatusCode(0);
```
Comment (alistairjevans) [inline tests/autostep-cli.tests/CommandInvocationTests.cs]: The underscores in the variable names here aren't consistent with the rest of the project.

**Review by alistairjevans — CHANGES_REQUESTED** [review]
This is great stuff! Just one tiny variable name change, and I'll merge this in.

**Comment by SivaGudivada** [comment]
Removed the underscore and updated the commits.

File: tests/autostep-cli.tests/CommandInvocationTests.cs [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+namespace AutoStep.CommandLine.Tests
+{
+    public class CommandInvocationTests
+    {
+        private static readonly CreateProjectDelegate successProjectCreation = MakeCreateProjectDelegateFromStatusCode(0);
+
+        private static CreateProjectDelegate MakeCreateProjectDelegateFromStatusCode(int returnStatusCode) =>
+            new CreateProjectDelegate((runArgs, iLogger) => returnStatusCode);
+
+        private static readonly AutoStepFiles _defaultMockAutoStepFiles = GetMockAutoStepFiles(successProjectCreation, successProjectCreation);
```
Comment (alistairjevans) [inline tests/autostep-cli.tests/CommandInvocationTests.cs]: One more variable name to fix.

Comment (SivaGudivada) [reply tests/autostep-cli.tests/CommandInvocationTests.cs]: My bad, clearly did not pay enough attention.

Comment (alistairjevans) [reply tests/autostep-cli.tests/CommandInvocationTests.cs]: No worries.

**Review by alistairjevans — APPROVED** [review]
Awesome; thanks for this, it's much appreciated. 😊

**Pull request merged**